Zend/zend_API.c

```diff
@@ -1960,82 +1960,67 @@ ZEND_API zend_module_entry* zend_register_internal_module(zend_module_entry *mod
 
 ZEND_API void zend_check_magic_method_implementation(const zend_class_entry *ce, const zend_function *fptr, int error_type) /* {{{ */
 {
-	char lcname[16];
-	size_t name_len;
+	zend_string *lcname;
 
 	if (ZSTR_VAL(fptr->common.function_name)[0] != '_'
 	 || ZSTR_VAL(fptr->common.function_name)[1] != '_') {
 		return;
 	}
 
-	/* we don't care if the function name is longer, in fact lowercasing only
-	 * the beginning of the name speeds up the check process */
-	name_len = ZSTR_LEN(fptr->common.function_name);
-	zend_str_tolower_copy(lcname, ZSTR_VAL(fptr->common.function_name), MIN(name_len, sizeof(lcname)-1));
-	lcname[sizeof(lcname)-1] = '\0'; /* zend_str_tolower_copy won't necessarily set the zero byte */
+	lcname = zend_string_tolower(fptr->common.function_name);
+	lcname = zend_new_interned_string(lcname);
```

Comment:
this change is a little unnecessary, even if you want to do, you need release this string, this must be a mem leak(at least in php life time)

Reply:
in zend_compile.c, the change is necessary is because the lc_name is also used in ce->function_table

Reply:
The zend_new_interned_string call here is unnecessary.

Reply:
Actually, you should be able to pass in the already lowercased method name for the two call sites of the function. For internal methods it's already available in lowercase_name, for userland it should be available if you move the check into begin_method (I don't think there's any reason for it not to be there).